modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala

```diff
@@ -56,9 +56,11 @@ object JacksonGenerator {
     parentOpt.foreach({ parent =>
       val directParent = JavaParser.parseClassOrInterfaceType(parent.clsName)
       val otherParents = parent.interfaces.map(JavaParser.parseClassOrInterfaceType)
-      cls.setExtendedTypes(
-        new NodeList((directParent +: otherParents): _*)
-      )
+      if (parent.discriminators.nonEmpty) {
```

Comment:
One caveat here is that this only checks to see if the discriminators are `nonEmpty` on the _first_ parent, hence `parentOpt`, `val parentOpt = parents.headOption`.

This deviates from the behaviour in the Circe generator, which checks _all_ `parents.exists(s => s.discriminators.nonEmpty)`

Recommendation is to consider the implications of this difference, then pick the one that will benefit users the most in the future.